test/cmdlineTests/~bytecode_equivalence_multiple_sources/test.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+from pathlib import Path
+
+PROJECT_ROOT = Path(__file__).parents[3]
+
+sys.path.insert(0, str(PROJECT_ROOT / "scripts"))
+#pylint: disable=wrong-import-position
+from common.cmdline_helpers import run_solc
+from common.cmdline_helpers import save_bytecode_report
+from common.cmdline_helpers import enter_tmptest_dir
+from common.git_helpers import diff
+from common.git_helpers import FileMismatchError
+from splitSources import split_sources
+
+
+@enter_tmptest_dir("cmdline-test-bytecode-equivalence-multiple-sources")
+def test_bytecode_equivalence():
+    tmp_dir = Path(os.getcwd())
+
+    source_file_path = Path(os.path.dirname(__file__)) / "inputs.sol"
+    split_sources(source_file_path, True)
+
+    # Compiling multiple files at same time should not affect bytecode generation although it changes AST IDs.
+    a_bytecode_path = tmp_dir / "A.bytecode"
+    a_report_output = run_solc(tmp_dir / "A.report", ["--via-ir", "--bin", "A.sol"])
+    save_bytecode_report("A", a_bytecode_path, a_report_output)
+
+    ab_bytecode_path = tmp_dir / "AB.bytecode"
+    ab_report_output = run_solc(tmp_dir / "AB.report", ["--via-ir", "--bin", "A.sol", "B.sol"])
+    save_bytecode_report("A", ab_bytecode_path, ab_report_output)
+
+    diff(a_bytecode_path, ab_bytecode_path)
+
+
+def main():
+    try:
+        test_bytecode_equivalence()
+        return 0
+    except (FileMismatchError, RuntimeError) as e:
+        print(f"Error: {e}", file=sys.stderr)
+        return 1
```

Comment:
I'd rather avoid having to include this handler in every test. Since no one is running this script directly, I think it's fine to just let these errors escape and interrupt the script.

I'd also remove `FileMismatchError`. Looks like your assumption is that non-zero exit means that there must have been differences in the content and that's not always true. It could just a well be an exit due to an error. If we're not taking an effort to properly distinguish it, I think it's better to just let the default exception escape. I'd just make the function always print its output (instead of returning it).

Overall, I'd prefer less boilerplate, even if it means a little less readable error output. As long as it fails on error and prints necessary details, it's good enough. I'd only put effort into making it more readable if the extra code needed that can be completely contained in the helpers, without having to be repeated in every test.

Reply:
Yes, this was my assumption. I just noticed that I removed the flag `--exit-code` (see: https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---exit-code). I used it in one of my initial attempts, we could use and then it would behave like `diff`, i.e.:
```
Make the program exit with codes similar to diff(1). That is, it exits with 1 if there were differences and 0 means no differences.
```

But yeah, I can remove the exception and just return the exit status.

Reply:
Please see https://github.com/ethereum/solidity/pull/14565/commits/31bedce7b476aff45339abd2f517e6bb13b68d2d :)